### PR TITLE
Add ScalaJS compilation

### PR DESCRIPTION
**Pull request by bijancn**
What remains to be done is add an explicit example to showcase that it really works. This could be done with an dockerCompose integration test (where we start up and shut down our custom server as docker image in the background during the test) or by just extending the example / adding a new example to be run by hand. The JS could be run in the browser or in the NodeJS runtime.

### CODE REVIEW

**Comment by pheymann** [comment]
Hi @bijancn 
thank you for the PR. I will have a look.

Regarding the showcase question: I would just update the examples for now.

File: server/src/main/scala/typedapi/server/RouteExtractor.scala [hunk 1]
```diff
@@ -48,7 +48,7 @@ trait RouteExtractorLowPrio {
 
     def apply(request: EndpointRequest, extractedHeaderKeys: Set[String], inAgg: EIn): Extract[Out] = request.uri match {
       case p :: tail => 
-        if (p == wit.value.toString())
```
Comment (pheymann) [inline server/src/main/scala/typedapi/server/RouteExtractor.scala]: Why did you remove the braces exclusively here?

Comment (pheymann) [reply server/src/main/scala/typedapi/server/RouteExtractor.scala]: Emacs is adding braces when interacting with Java methods and I never bothered to change that :). Therefore, you will find `toString()` multiple times in this code base.

Comment (bijancn) [reply server/src/main/scala/typedapi/server/RouteExtractor.scala]: IntelliJ marks expressions with unnecessary braces yellow. And I dont like warnings ;). But in this case they will actually become necessary with dotty I think. This makes it possible to refer to the function itself without any extra `_`

**Comment by pheymann** [comment]
👍
